sn_protocol/src/storage/header.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::error::{Error, Result};
+use libp2p::kad::Record;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RecordHeader {
+    pub kind: RecordKind,
+}
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+pub enum RecordKind {
+    Chunk,
+    DbcSpend,
+    Register,
+}
+
+impl Display for RecordKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "RecordKind({self:?})")
+    }
+}
+
+impl RecordHeader {
+    // Bincode serializes enums with unit variants as a u32, not a u8, so it would take up 4 bytes.
```

Comment:
I don't think we can use bincode here. The protocol should be language agnostic and `bincode` is rust specific

Reply:
Updated the PR to use `rmp_serde` instead.